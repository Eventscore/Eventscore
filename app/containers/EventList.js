import React, { Component } from 'react';
import {
  Image,
  ListView,
  TouchableHighlight,
  StyleSheet,
  Text,
  View,
  ActivityIndicator,
  RefreshControl,
  ScrollView,
} from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { ActionCreators } from '../actions';
import EventListItem from './EventListItem';
import EventListMap from './EventListMap';
// import NavBar from './NavBar';
import TabBar from './TabBar';
import Search from './Search'; //remove this later, only for testing

class EventList extends Component {
  static title = '<RefreshControl>';
  static description = 'Adds pull-to-refresh support to a scrollview.';
  constructor() {
    super();
    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
    this.state = {
      eventList: ds.cloneWithRows([]),
      listSwitch: false,
      isRefreshing: false,
      loaded: 0,
    };
  }

  searchPressed() {
    this.getNearbyEvents();
  }

  async getNearbyEvents() {
    let getLocation = await this.props.getLocation();
    this.props.fetchNearbyEvents(
      this.props.locationReducers.geolocation.coords.longitude,
      this.props.locationReducers.geolocation.coords.latitude
    ).then(() => {
      this.setState({
        eventList: this.state.eventList.cloneWithRows(this.props.eventsReducers.events),
      });
    }).catch((error) => {
      console.log(error);
    });
  }

  componentWillMount() {
  }

  componentDidMount() {
    if (!this.props.eventsReducers.events) {
      this.getNearbyEvents();
    } else {
      this.setState({
          eventList: this.state.eventList.cloneWithRows(this.props.eventsReducers.events),
      });
    }
  }

  render() {
    let list = null;
    if (this.state.listSwitch) {
      list = <EventListMap />;
    } else {
      list = 
        <View style={styles.eventContainer}>
          <ScrollView
          refreshControl={
            <RefreshControl
              refreshing={this.state.isRefreshing}
              onRefresh={() => this.searchPressed()}
              tintColor="#ff0000"
              title="Loading..."
              titleColor="#00ff00"
              colors={['#ff0000', '#00ff00', '#0000ff']}
              progressBackgroundColor="#ffff00"
            />
          }>
          <ListView
            dataSource={this.state.eventList}
            renderRow={(event) => <EventListItem key={event._id} event={event} />}
          />
          </ScrollView>
        </View>;
    }

    let listPageRender = null;
    if (this.props.eventsReducers.loadingEvents) {
      listPageRender = 
        <View style={styles.eventContainer}>
          <ActivityIndicator size='large' style={{height: 80}} />
        </View>;
    } else if (this.props.eventsReducers.cannotGetEvents) { // if cannot get user geolocation
      listPageRender = 
        <View style={styles.eventContainer}>
          <TouchableHighlight onPress={ () => this.searchPressed() }>
            <Text style={styles.fetchEventsText}>Check Nearby Events!</Text>
          </TouchableHighlight>
          <Text style={{textAlign: 'center'}}>Error, please try again</Text>
        </View>;
    } else if (this.props.eventsReducers.events) {
      listPageRender = 
        <View style={styles.eventContainer}>
          {list}
            <TouchableHighlight style={styles.floatButtonContainer} onPress={ (value) => { this.setState({listSwitch: !this.state.listSwitch}); }} value={this.state.listSwitch}>
              <Text style={{fontSize: 15, color: '#FFF'}}>{this.state.listSwitch === false ? 'Map' : 'List'}</Text>
            </TouchableHighlight>
        </View>;
    } else {
      listPageRender = 
        <View style={styles.eventContainer}>
        </View>;
    }

    return (
      <View style={{flex: 1}}>
        <View style={{flex: -1, zIndex: 1}}>
          <Search />
        </View>
        <View style={{flex: 8, zIndex: 0}}>
          {listPageRender}
        </View>
        <View style={{flex: 1, zIndex: 2}}>
          <TabBar />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  fetchEventsText: {
    fontSize: 20,
    fontWeight: 'bold',
    textAlign: 'center',
    color: '#333333',
    margin: 5
  },
  container: {
    flex: 1,
    color: '#000000'
  },
  eventContainer: {
    flex: 1,
    backgroundColor: '#050505',
    justifyContent: 'center',
    display: 'flex',
  },
  switch: {
    marginTop: 5,
    fontSize: 20,
    color: '#FFF'
  },
  switchContainer: {
    alignItems: 'center'
  },
  floatButtonContainer: {
    backgroundColor: '#ff5722',
    borderColor: '#ff5722',
    borderWidth: 1,
    height: 50,
    width: 50,
    borderRadius: 50,
    alignItems: 'center',
    justifyContent: 'center',
    position: 'absolute',
    bottom: 20,
    right:20,
    shadowColor: "#000000",
    shadowOpacity: 0.8,
    shadowRadius: 2,
    shadowOffset: {
      height: 1,
      width: 0
    }
  }
});

function mapDispatchToProps(dispatch) {
  return bindActionCreators(ActionCreators, dispatch);
}

export default connect(({routes, loginReducers, eventsReducers, locationReducers}) => { return {routes, loginReducers, eventsReducers, locationReducers}}, mapDispatchToProps)(EventList);