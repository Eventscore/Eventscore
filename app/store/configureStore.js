//Uncertain if this will work
// import { createStore, applyMiddleware, combineReduxers, compose } from 'redux';
// import thunkMiddleware from 'redux-thunk';
// import reducers from '../reducers';
// import { createLogger } from 'redux-logger';

// //this logs only in development mode
// const loggerMiddleware = createLogger( { predicate: (getState, action) => __DEV__ });

// export default function configureStore() {
//   const enhancers = compose(
//     applyMiddleware(
//       thunkMiddleware,
//       loggerMiddleware,
//     ),
//   );

//   const store = createStore(reducers, enhancers);

//   if(module.hot) {
//     module.hot.accept( () => {
//       const nextRootReducer = require('../reducers/index').default;
//       store.replaceReducer(nextRootReducer);
//     });
//   }
//   return store;
// }

import { createStore, applyMiddleware, compose } from 'redux';
import reducers from '../reducers/index';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';

const loggerMiddleware = createLogger({ predicate: (getState, action) => __DEV__ })

export default function configureStore (initialState) {
  // const middleware = [thunkMiddleware, loggerMiddleware];
  const enhancer = compose(
    applyMiddleware(
      thunkMiddleware,
      loggerMiddleware
    )
  )
  const store = createStore(reducers, initialState, enhancer);
  // const store = compose(
  //   applyMiddleware(...middleware))(createStore)(reducers);

  // if (module.hot) {
  //   module.hot.accept(() => {
  //     const nextRootReducer = require('../reducers/index').default
  //     store.replaceReducer(nextRootReducer)
  //   })
  // }
  return store;
}

