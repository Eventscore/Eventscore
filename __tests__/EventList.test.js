import 'react-native';
import React from 'react';
import EventList from '../app/containers/EventList';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

const initialState = {
  eventsReducers: {
    loadingEvents: false
  },
  listTypeReducers: {
    listType: false
  }
};

const mockStore = configureStore(initialState);
const store = mockStore(initialState);

it('EventList renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}><EventList /></Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});