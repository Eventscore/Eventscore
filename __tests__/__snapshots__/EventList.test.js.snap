// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventList renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": -1,
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": -1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": -1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#000000",
              "borderBottomColor": "#b5b5b5",
              "borderBottomWidth": 0,
              "borderTopColor": "#7e7e7e",
              "borderTopWidth": 0,
              "height": 44,
              "width": 750,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginTop": 25,
                "textAlign": "center",
              }
            }
          >
            EventScore
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#000000",
              "borderBottomColor": "#b5b5b5",
              "borderBottomWidth": 0,
              "borderTopColor": "#7e7e7e",
              "borderTopWidth": 0,
              "flexDirection": "row",
              "height": 44,
              "width": 675,
            }
          }
        >
          <TextInput
            onChangeText={[Function]}
            placeholder="e.g Beyonce, The Weeknd, Bruno Mars"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 5,
                "flex": 1,
                "fontSize": 15,
                "height": 28,
                "marginLeft": 8,
                "marginRight": 8,
                "marginTop": 7.5,
                "paddingBottom": 4.5,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 4.5,
              }
            }
            value={undefined}
          />
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#C9C9CE",
                  "borderBottomColor": "#b5b5b5",
                  "borderBottomWidth": 0.5,
                  "borderTopColor": "#7e7e7e",
                  "borderTopWidth": 0.5,
                  "flexDirection": "row",
                  "height": 44,
                },
                Object {
                  "backgroundColor": "#000000",
                  "borderBottomColor": "#b5b5b5",
                  "borderBottomWidth": 0,
                  "borderTopColor": "#7e7e7e",
                  "borderTopWidth": 0,
                  "flexDirection": "row",
                  "height": 44,
                  "width": 675,
                },
              ]
            }
          >
            <TextInput
              autoFocus={false}
              clearButtonMode="while-editing"
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Search"
              placeholderTextColor="#A8A8A8"
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 5,
                    "flex": 1,
                    "fontSize": 15,
                    "height": 28,
                    "marginLeft": 8,
                    "marginRight": 8,
                    "marginTop": 7.5,
                    "paddingBottom": 4.5,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 4.5,
                  },
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 5,
                    "color": "blue",
                    "flex": 1,
                    "fontSize": 15,
                    "height": 28,
                    "marginLeft": 8,
                    "marginRight": 8,
                    "marginTop": 7.5,
                    "paddingBottom": 4.5,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 4.5,
                  },
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": -1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#000000",
              "borderBottomColor": "#b5b5b5",
              "borderBottomWidth": 0,
              "borderTopColor": "#7e7e7e",
              "borderTopWidth": 0,
              "height": 44,
              "justifyContent": "center",
              "width": 75,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#000000",
              "borderBottomColor": "#b5b5b5",
              "borderBottomWidth": 0,
              "borderTopColor": "#7e7e7e",
              "borderTopWidth": 0,
              "height": 88,
              "justifyContent": "center",
              "width": 75,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 8,
        "zIndex": 0,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "borderStyle": "solid",
          "borderTopColor": "#fff",
          "borderTopWidth": 2,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#282828",
            "display": "flex",
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-around",
            "padding": 5,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={Object {}}
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "display": "flex",
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                resizeMode="contain"
                style={
                  Array [
                    Object {
                      "color": "#FFF",
                      "fontSize": 30,
                    },
                    Object {},
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                  }
                }
              >
                Home
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={Object {}}
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hitSlop={undefined}
              isTVSelectable={true}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "display": "flex",
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                resizeMode="contain"
                style={
                  Array [
                    Object {
                      "color": "#FFF",
                      "fontSize": 30,
                    },
                    Object {},
                    Object {
                      "fontFamily": "FontAwesome",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                  }
                }
              >
                Events
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
